src/oprf.ts

```diff
@@ -137,83 +144,68 @@ export class Evaluation {
         if ((this.proof && !e.proof) || (!this.proof && e.proof)) {
             return false
         }
-        let res = this.evaluated.isEqual(e.evaluated)
+        let res = this.evaluated.every((x, i) => x.isEqual(e.evaluated[i as number]))
         if (this.proof && e.proof) {
             res &&= this.proof.isEqual(e.proof)
         }
         return res
     }
 
-    static size(params: DLEQParams): number {
-        return Elt.size(params.gg) + 1
-    }
-
     static deserialize(params: DLEQParams, bytes: Uint8Array): Evaluation {
-        checkSize(bytes, Evaluation, params)
-        const eltSize = Elt.size(params.gg)
-        const evaluated = Elt.deserialize(params.gg, bytes.subarray(0, eltSize))
+        const { head: evalList, tail } = fromU16LenPrefixClass(Elt, params.gg, bytes)
         let proof: DLEQProof | undefined
-        if (bytes[eltSize as number] === 1) {
+        if (tail[0] === 1) {
```

Comment:
Making this `1` into a constant might make it more clear that it's an indicator of whether or not the proof is included (see other comment).